src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -52,10 +58,56 @@ class RegularSync(
       askForHeaders()
   }
 
-  def running: Receive = handleCommonMessages orElse {
-    case ResumeRegularSync =>
-      askForHeaders()
+  private def resumeRegularSync(): Unit = {
+    resumeRegularSyncTimeout.foreach(_.cancel)
+    resumeRegularSyncTimeout = None
+    askForHeaders()
+  }
 
+  /**
+    * Handles broadcasted blocks, processing them and inserting them to the blockchain if necessary
+    * FIXME: Current implementation is naive in that NewBlock msgs that can't be immediately processed are discarded
+    *        and requested again to the peers
+    */
+  def handleBroadcastedBlockMessages: Receive = {
```

Comment:
Minor: I found this function with all its comments, cases, `_` marks really hard to decipher what is the intent here. It maybe just my opinion as I am still new to this regular sync, so feel free ignore this comment. (maybe as the bigger refactor of regular sync is coming it not have much sense)

Reply:
I have a similar impression. Maybe at least adding some vertical spacing would improve readability?